packages/tiled/lib/src/common/property.dart

```diff
@@ -33,14 +33,21 @@ class Property {
 }
 
 extension PropertiesParser on Parser {
-  List<Property> getProperties() {
-    return formatSpecificParsing(
+  Map<String, Property> getProperties() {
+    final properties = formatSpecificParsing(
       (json) => json.getChildrenAs('properties', Property.parse),
       (xml) =>
           xml
               .getSingleChildOrNull('properties')
               ?.getChildrenAs('property', Property.parse) ??
           [],
     );
+
+    return properties.groupFoldBy((prop) => prop.name, (previous, element) {
+      if (previous != null) {
+        throw ArgumentError("Can't have two properties with the same name.");
```

Comment:
I don't think we should throw an error here. If `Tiled` does not even allow creation of multiple properties with same name, this case should ideally never occur. Our input condition is `a valid tmx file`.

In the rare case someone manually tweaks their tmx and adds multiple properties with same name, instead of failing, we should follow what Tiled does. Tiled load the map and displays only one of the duplicate properties (the latest that it finds). We can also do that by replacing the value with latest value with a duplicate key is found.

If we really want to go the extra mile and inform the users, it can be done using an assert.